src/explorer.api/Controllers/ExploreController.cs

```diff
@@ -126,33 +120,47 @@ public async Task<IActionResult> Result(System.Guid explorationId)
             }
         }
 
+        [HttpGet]
+        [Route("cancel/{explorationId}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public IActionResult Cancel(System.Guid explorationId)
+        {
+            if (!Explorations.TryGetValue(explorationId, out var exploration))
+            {
+                return BadRequest($"Couldn't find exploration with id {explorationId}");
+            }
+            exploration.Cancel();
+            return Ok();
+        }
+
         [Route("/{**catchall}")]
         [ProducesResponseType(StatusCodes.Status404NotFound)]
         public IActionResult OtherActions() => NotFound();
 
-        private Exploration? CreateExploration(AircloakType type, Models.ExploreParams data)
+        private Exploration? CreateExploration(AircloakType type, Models.ExploreParams data, CancellationTokenSource cts)
         {
-            var resolver = new AircloakQueryResolver(apiClient, data.DataSourceName);
+            var resolver = new AircloakQueryResolver(apiClient, data.DataSourceName, config.PollFrequencyTimeSpan);
 
             var components = type switch
             {
                 AircloakType.Integer => new ExplorerBase[]
                 {
-                    new IntegerColumnExplorer(resolver, data.TableName, data.ColumnName),
-                    new MinMaxExplorer(resolver, data.TableName, data.ColumnName),
+                    new IntegerColumnExplorer(resolver, data.TableName, data.ColumnName, cts.Token),
+                    new MinMaxExplorer(resolver, data.TableName, data.ColumnName, cts.Token),
                 },
```

Comment:
Ok.. I can see how creating the cts.Token in the Exploration might be tricky. 

We may have to have to move this switch statement into the `Exploration` constructor, otherwise we can pass delegates instead of Explorer instances. Something like: 
```csharp
var components = (token) => new ExplorerBase[]
                {
                    new IntegerColumnExplorer(resolver, data.TableName, data.ColumnName, token),
                    new MinMaxExplorer(resolver, data.TableName, data.ColumnName, token),
                },

[etc.]
```

and in the Exlporation constructor:
```csharp
Exploration(Func<Token, ExplorerBase[]> explorerBuilder)
{
    var tokenSource = new CancellationTokenSource();
    var components = explorerBuilder(tokenSource.Token);
[etc.]
}
```

Reply:
fixed with 60fa1549b948208e2d7edcecb13fbf6bd2a83179